devito/types/dimension.py

```diff
@@ -520,19 +532,27 @@ def middle(cls, name, parent, thickness_left, thickness_right, local=False):
                    thickness=((lst, thickness_left), (rst, thickness_right)),
                    local=local)
 
-    @property
+    @cached_property
```

Comment:
Caching these is not going to cause problems when we're updating them via implicit equations right?

Reply:
it must not. These objects are immutable